src/operators/ai.onnx/ConvTranspose/11/execute_operator__ai_onnx__convtranspose__11__T_tensor_float.c

```diff
@@ -0,0 +1,193 @@
+#include "tracing.h"
+#include "utils.h"
+
+//defined in prepare
+int calcOutputSize(int inputSize, int kernelSize, int stride, int dilations, int padStart, int padEnd);
+
+//transformes a 2d pos into a 1d flat array pos
+int calcArrayPos2D(int x, int y, int width) {
+    return y * width + x;
+}
+
+//transformes a 3d pos into a 1d flat array pos
+int calcArrayPos3D(int x, int y, int outputChannel, int width, int height) {
+    return outputChannel * height * width + y * width + x;
+}
+
+//transformes a 4d pos into a 1d flat array pos
+int calcArrayPos4D(int x, int y, int outputChannel, int intputChannel, int width, int height, int nOfOutputChannels) {
+    return intputChannel * nOfOutputChannels *height * width 
+           + outputChannel * height * width 
+           + y * width
+           + x;
+}
+
+operator_status
+execute_operator__ai_onnx__convtranspose__11__T_tensor_float(
+    node_context *ctx
+)
+{
+    TRACE_ENTRY(1);
+
+    TRACE_NODE(2, true, ctx->onnx_node);
+
+    /* UNCOMMENT AS NEEDED */
+
+    Onnx__TensorProto *i_X = searchInputByName(ctx, 0);
+    Onnx__TensorProto *i_W = searchInputByName(ctx, 1);
+    Onnx__TensorProto *i_B = searchInputByName(ctx, 2);
+
+    // TRACE_TENSOR(2, true, i_X);
+    // TRACE_TENSOR(2, true, i_W);
+    // TRACE_TENSOR(2, B, i_B);
+    Onnx__AttributeProto *a_dilations = searchAttributeNyName(ctx->onnx_node->n_attribute,ctx->onnx_node->attribute,"dilations");
+    Onnx__AttributeProto *a_pads = searchAttributeNyName(ctx->onnx_node->n_attribute,ctx->onnx_node->attribute,"pads");
+    Onnx__AttributeProto *a_strides = searchAttributeNyName(ctx->onnx_node->n_attribute,ctx->onnx_node->attribute,"strides");
+
+    // context_operator__ai_onnx__convtranspose__11 *op_ctx = ctx->executer_context;
+
+    // char* auto_pad = op_ctx->auto_pad;
+    // size_t n_dilations = op_ctx->n_dilations;
+    // int64_t* dilations = op_ctx->dilations;
+    // int64_t group = op_ctx->group;
+    // size_t n_kernel_shape = op_ctx->n_kernel_shape;
+    // int64_t* kernel_shape = op_ctx->kernel_shape;
+    // size_t n_output_padding = op_ctx->n_output_padding;
+    // int64_t* output_padding = op_ctx->output_padding;
+    // size_t n_output_shape = op_ctx->n_output_shape;
+    // int64_t* output_shape = op_ctx->output_shape;
+    // size_t n_pads = op_ctx->n_pads;
+    // int64_t* pads = op_ctx->pads;
+    // size_t n_strides = op_ctx->n_strides;
+    // int64_t* strides = op_ctx->strides;
+
+    // TRACE_VAR(2, true, auto_pad, "\"%s\"");
+    // TRACE_ARRAY(2, true, dilations, , n_dilations, "%" PRId64);
+    // TRACE_VAR(2, true, group, "%" PRId64);
+    // TRACE_ARRAY(2, true, kernel_shape, , n_kernel_shape, "%" PRId64);
+    // TRACE_ARRAY(2, true, output_padding, , n_output_padding, "%" PRId64);
+    // TRACE_ARRAY(2, true, output_shape, , n_output_shape, "%" PRId64);
+    // TRACE_ARRAY(2, true, pads, , n_pads, "%" PRId64);
+    // TRACE_ARRAY(2, true, strides, , n_strides, "%" PRId64);
```

Comment:
also unused

Reply:
Should I prefer op_ctx over searchAttributeNyName?

Reply:
the best way would be to resolve attributes once with `searchAttributeNyName` and save them to `op_ctx` (or replace them with a default) so the executer doesn't have to resolve these again.

Reply:
Now i get it. I will adopt my implementation